ansible_collections/studyly/git_mactp/plugins/module_utils/git_actions.py

```diff
@@ -183,20 +208,44 @@ def commit(self):
             }
         FailingMessage(self.module, rc, command, output, error)
 
+    def merge(self):
+        """
+        Merge into checked out branch.
+
+        args:
+            * module:
+                type: dict()
+                description: Ansible basic module utilities and module arguments.
+        return:
+            * result:
+                type: dict()
+                description: returned output from git merge command and changed status
+        """
+        source_branch = self.module.params.get("merge")
+        merge_options = self.module.params.get("merge_options")
+
+        command = [self.git_path, "merge", source_branch]
+
+        if merge_options:
+            for opt in merge_options:
+                command.insert(2, opt)
+
+        rc, output, error = self.module.run_command(command, cwd=self.path)
+
+        if rc == 0:
+            return {
+                "git_merge": {"output": output, "error": error, "changed": True}
```

Comment:
As discussed, a merge might not actually update a branch. We should somehow evaluate the output. The simplest I can think of would be to check if the output contained something like "everything up to date" (or whatever it was exactly). To do this right, we'd have to ensure the locale on the machine where this is run is en_US or en_GB, but this seems to be a limitation of the module already, thus I'd be okay with simply assuming the locale.